java-frontend/src/main/java/org/sonar/java/model/SyntacticEquivalence.java

```diff
@@ -62,27 +65,27 @@ public static boolean areEquivalent(List<? extends Tree> leftList, List<? extend
    * @return true, if nodes are syntactically equivalent
    */
   public static boolean areEquivalent(@Nullable Tree leftNode, @Nullable Tree rightNode) {
-    return areEquivalent(leftNode, rightNode, (t1, t2) -> false);
+    return areEquivalent(leftNode, rightNode, (t1, t2) -> false, true);
   }
 
   /**
    * @return true, if nodes are syntactically equivalent
-   * Use permissiveEquivalence to force the equivalence of two nodes
+   * Use "overwriteEquivalence" to force the equivalence or not of two nodes. When it returns true, the method will return "equivalenceValue".
```

Comment:
What about "overwritten/overridden"
and
"to force the equivalence check result"